### PR TITLE
[plugin.video.catchuptvandmore@krypton] 0.2.38

### PR DESCRIPTION
**Pull request by joaopa00**
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Once more, this PR is pretty huge. If this PR is merged very quickly, I will send the newest version very soon. So, the PR for the next coming version 0.2.39 should be relatively small, so easier to review.

Thanks for your understanding.

### CODE REVIEW

**Comment by joaopa00** [comment]
What means the error in Kodi addon-checker?

**Comment by enen92** [comment]
Means you need higher versions in matrix and nexus for users with this version installed in krypton to be able to receive the update when migrating to an up to date Kodi version. 
If you already have those updates pending (for matrix and nexus) ignore the message, ci will pass once the other two prs are merged.

**Comment by joaopa00** [comment]
I see. Since I already sent the version the version for matrix, there is no problem I think.
Can you confirm?

**Comment by joaopa00** [comment]
I removed the problematic feature. Thank you for the review and the feedback.

(Not so) unrelated note: I pushed some logos and picture in the repository repo-resources. Can you merge them?

**Comment by joaopa00** [comment]
I removed the problematic feature. Thank you for the review and the feedback.

(Not so) unrelated note: I pushed some logos and picture in the repository repo-resources. Can you merge them?

File: plugin.video.catchuptvandmore/resources/lib/enum/LICENSE [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Copyright (c) 2013, Ethan Furman.
```
Comment (pkscout) [inline plugin.video.catchuptvandmore/resources/lib/enum/LICENSE]: this is the same issue I flagged in the Matrix version.  This module needs to be submitted separately in a Kodi addon wrapper.

**Review by pkscout — CHANGES_REQUESTED** [review]

**Comment by joaopa00** [comment]
I sent a PR to repo-scripts to merge enum34. Once that's done, I shall update this PR to take advantage of this feature.

**Comment by joaopa00** [comment]
I removed the dependency on Enum.
Please, have a look at this new version

**Comment by joaopa00** [comment]
I modified this PR following the recommendations from the Matrix PR versions.
Please have a look.

**Comment by basrieter** [comment]
I will double check it.

**Comment by joaopa00** [comment]
Any news for this PR will be welcome. Thanks in advance for any feedback

**Comment by joaopa00** [comment]
Is there anything more I should do to make this pull request merged?

**Comment by enen92** [comment]
@joaopa00 check addon checker log, looks like you have some incorrectly closed ")" somewhere. Might check the execution locally to see where that is exactly triggered

**Comment by connesc** [comment]
@pkscout @enen92, sorry to bother you, but the status of this PR is not really clear to me.
Are there blocking points remaining? Any chance to see this merged soon?

**Comment by enen92** [comment]
Yes, pull request is completely broken...

**Comment by joaopa00** [comment]
How can I know where is the problem with local kodi-addon-checker?

$ kodi-addon-checker --branch krypton   --enable-debug-log
INFO: Checking add-on plugin.video.catchuptvandmore
INFO: Created by SylvainCecchetto,wwark,joaopa
INFO: Addon id matches folder name
INFO: Valid XML file found
INFO: Image icon exists
INFO: icon dimensions are fine 512x512
INFO: Image fanart exists
INFO: fanart dimensions are fine 1280x720
INFO: fanart file size is fine 690KB
INFO: Image screenshot exists
INFO: screenshot dimensions are fine 1920x1080
INFO: screenshot file size is fine 124KB
INFO: Image screenshot exists
INFO: screenshot dimensions are fine 1920x1080
INFO: screenshot file size is fine 452KB
INFO: Image screenshot exists
INFO: screenshot dimensions are fine 1920x1080
INFO: screenshot file size is fine 114KB
INFO: Image screenshot exists
INFO: screenshot dimensions are fine 1920x1080
INFO: screenshot file size is fine 339KB
INFO: Image screenshot exists
INFO: screenshot dimensions are fine 1920x1080
INFO: screenshot file size is fine 229KB
ERROR: ParseError: bad input: type=8, value=')', context=('', (332, 48))
INFO: ParseError: bad input: type=8, value=')', context=('', (332, 48))
INFO: PO files are valid
WARN: Found non whitelisted file ending in filename ./plugin.video.catchuptvandmore/kodi-addon-checker.log
ERROR: We found 1 problems and 1 warnings, please check the logfile.

**Comment by enen92** [comment]
That means you have a syntax error in your python code some parenthesis that don't have a matching close tag. I can't really tell where it is with that log though.
Also please drop any github actions file from the submission

**Comment by joaopa00** [comment]
Fortunately, pylint is better than kodi-addon-checker to explicit syntax errors.  I also fixed the problem you pinpointed about github action.

File: plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py [hunk 2]
```diff
@@ -236,6 +236,7 @@ def list_videos_search(plugin, search_query, item_id, page, **kwargs):
                         item_post_treatment(item,
                                             is_playable=True,
                                             is_downloadable=is_downloadable)
+                        yield
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py]: this looks like a leftover? You yield the actual item below

File: plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py [hunk 3]
```diff
@@ -209,9 +209,9 @@ def list_videos_search(plugin, search_query, item_id, page, **kwargs):
                                       item_id=item_id,
                                       program_id=search_id)
                     item_post_treatment(item)
+                    yield
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py]: looks like a leftover, you yield the item in the line below

File: plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py (hunk 2 above)
Comment (joaopa00) [reply plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py]: good catch. I fixed that

File: plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py (hunk 3 above)
Comment (joaopa00) [reply plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py]: Thanks you again

**Review by enen92 — APPROVED** [review]

**Comment by enen92** [comment]
```
File "/opt/hostedtoolcache/Python/3.11.2/x64/lib/python3.11/lib2to3/pgen2/driver.py", line 47, in parse_tokens
    for quintuple in tokens:
  File "/opt/hostedtoolcache/Python/3.11.2/x64/lib/python3.11/lib2to3/pgen2/tokenize.py", line 424, in generate_tokens
    raise IndentationError(
  File "<tokenize>", line 2[38](https://github.com/xbmc/repo-plugins/actions/runs/4336770275/jobs/7572297642#step:6:39)
    yield item
IndentationError: unindent does not match any outer indentation level
```

**Comment by joaopa00** [comment]
Thanks. I fixed this problem too.

File: plugin.video.catchuptvandmore/resources/lib/enum/LICENSE (hunk 1 above)
Comment (joaopa00) [reply plugin.video.catchuptvandmore/resources/lib/enum/LICENSE]: Done. Thanks fir the feedback

**Comment by joaopa00** [comment]
Thank you for merging this PR. Sorry for the trouble. Is it possible to merge the pictures from repo-ressource?

**Pull request merged**